storage/engine/encoding.cc

```diff
@@ -24,16 +24,37 @@ const unsigned char kEscape      = 0x00;
 const unsigned char kEscapedTerm = 0x01;
 const unsigned char kEscapedNul  = 0xff;
 
+template <typename T>
+bool DecodeVarint(rocksdb::Slice* buf, T* value) {
+  if (buf->empty()) {
+    return false;
+  }
+  int len = (*buf)[0];
+  if ((len + 1) > buf->size()) {
```

Comment:
Doesn't hurt. Note that this code was pulled verbatim from the viewfinder sources.